### PR TITLE
DPC-1260: The implementer portal calls the API to create a new implementer

### PR DESCRIPTION
**Pull request by Sun-Mountain**
### Fixes [DPC-1260](https://jira.cms.gov/browse/DPC-1260)

When a user submits the account creation form, the rails app should call the API to create a new implementer.

### Proposed Changes

- If `implementer_id` blank, kick off `create_api_imp` (this ensures it will not kick off for an invited user account creation)
- In user model: `create_api_imp` method
- Create `ApiClient` service
- In tests:
  - Stubbed existing tests to ensure passing
  - Created tests for `ApiClient` service
  - Created ApiClientSupport for easier stubbing

### Change Details

- Changed `make start portals` to use the dpc-v2 postgres database
- Added README instructions to set up `.env` files to run tests
- Added `ApiErrorSimplify` concern to make errors more readable
- In `http_request` method, added `rescue` for catching `Errno::ECONNREFUSED` and returning a usable error

### Security Implications

- [ ] new software dependencies

- [ ] security controls or supporting software altered

- [ ] new data stored or transmitted

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

- [x] no PHI/PII is affected by this change

### Acceptance Validation

- When a user submits the account creation form, the rails app should call the API to create a new implementer.
- Implementer creation is automatic. Admin users do NOT need to approve anything.
- The Implementer ID is stored in the Rails database

Screen Shot 2021-05-27 at 2 12 51 PM

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -49,15 +49,15 @@ start-local-api: secure-envs start-local
 
 .PHONY: start-portals
 start-portals:
-	@docker-compose -p dpc-portals -f docker-compose.yml -f docker-compose.portals.yml up start_core_dependencies
-	@docker-compose -p dpc-portals -f docker-compose.yml -f docker-compose.portals.yml up start_web
-	@docker-compose -p dpc-portals -f docker-compose.yml -f docker-compose.portals.yml up start_admin
-	@docker-compose -p dpc-portals -f docker-compose.yml -f docker-compose.portals.yml up start_impl
+	@docker-compose -p dpc-v2 -f docker-compose.yml -f docker-compose.portals.yml up start_core_dependencies
+	# @docker-compose -p dpc-v2 -f docker-compose.yml -f docker-compose.portals.yml up start_web
+	# @docker-compose -p dpc-v2 -f docker-compose.yml -f docker-compose.portals.yml up start_admin
```
Comment (dhgreene) [inline Makefile:54]: If we don't need the v1 web and admin Rails apps to run at the same time as the v2 portal(s), we should:

- Consider renaming the Make target to `start-v2-portals` (maybe?)
- Delete the commented-out lines
- Update the lists in the `dpc-admin/README.md` lines 90-92
- Say something else in `dpc-admin/README.md` lines 72-75

**Review by dhgreene — COMMENTED** [review]
Would you mind building this branch in Jenkins?

File: dpc-impl/config/database.yml [hunk 2]
```diff
@@ -46,7 +46,7 @@ development:
 
   # The TCP port the server listens on. Defaults to 5432.
   # If your server runs on a different port number, change accordingly.
-  #port: 5432
+  port: 2345
```
[the remark this answers is not in the document]
Comment (Sun-Mountain) [reply dpc-impl/config/database.yml:49]: Oh I didn't update this branch with that port.

File: Makefile (hunk 1 above)
Comment (Sun-Mountain) [reply Makefile:54]: - Changed the Make target prefix to `start-v2-portals`

- The commented out lines were to make running the impl tests faster. They aren't supposed to stay commented out. Those run the tests for the web and admin apps. Uncommented those tests.

Although, some web and admin tests are failing locally (but passing in Github actions). So maybe that's something I can look into next sprint.

- Updated lists in `dpc-admin/README.md` lines 90-92
- Deleted 72-75 lines in `dpc-admin/README.md`

**Review by dhgreene — APPROVED** [review]
Looks good! 🛤️

**Pull request merged**